### ISSUE TITLE
Server not starting after updating ancientwarfare

### ISSUE THREAD

**Issue by wadewegner** [P3pp3rF1y/AncientWarfare2#739]
Hey there @P3pp3rF1y! Long time.

I tried to update my world today. Crashing:

```
[21:20:33] [Server thread/INFO]: Starting minecraft server version 1.12.2
[21:20:33] [Server console handler/ERROR]: Exception handling console input
java.io.IOException: Bad file descriptor
	at java.io.FileInputStream.readBytes(Native Method) ~[?:1.8.0_171]
	at java.io.FileInputStream.read(FileInputStream.java:255) ~[?:1.8.0_171]
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:284) ~[?:1.8.0_171]
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_171]
	at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284) ~[?:1.8.0_171]
	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326) ~[?:1.8.0_171]
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178) ~[?:1.8.0_171]
	at java.io.InputStreamReader.read(InputStreamReader.java:184) ~[?:1.8.0_171]
	at java.io.BufferedReader.fill(BufferedReader.java:161) ~[?:1.8.0_171]
	at java.io.BufferedReader.readLine(BufferedReader.java:324) ~[?:1.8.0_171]
	at java.io.BufferedReader.readLine(BufferedReader.java:389) ~[?:1.8.0_171]
	at net.minecraft.server.dedicated.DedicatedServer$2.run(DedicatedServer.java:105) [nz$2.class:?]
[21:20:59] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.NoSuchMethodError: net.minecraftforge.fml.common.registry.GameRegistry.registerTileEntity(Ljava/lang/Class;Lnet/minecraft/util/ResourceLocation;)V
	at net.shadowmage.ancientwarfare.structure.block.AWStructuresBlockLoader.registerTile(AWStructuresBlockLoader.java:72) ~[AWStructuresBlockLoader.class:?]
	at net.shadowmage.ancientwarfare.structure.block.AWStructuresBlockLoader.register(AWStructuresBlockLoader.java:50) ~[AWStructuresBlockLoader.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_20_AWStructuresBlockLoader_register_Register.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus$1.invoke(EventBus.java:143) ~[EventBus$1.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:179) ~[EventBus.class:?]
	at net.minecraftforge.registries.GameData.fireRegistryEvents(GameData.java:734) ~[GameData.class:?]
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:604) ~[Loader.class:?]
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:98) ~[FMLServerHandler.class:?]
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:332) ~[FMLCommonHandler.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:125) ~[nz.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:484) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
[21:20:59] [Server thread/ERROR]: This crash report has been saved to: /home/wadewegner/minecraft-1.12.2/./crash-reports/crash-2018-07-15_21.20.59-server.txt
[21:20:59] [Server thread/INFO]: Stopping server
[21:20:59] [Server thread/INFO]: Saving worlds
[1]+  Done                    nohup java -Xmx1024M -Xms1024M -jar forge-1.12.2-14.23.1.2604-universal.jar nogui > /dev/null 2>&1
```

`cat /home/wadewegner/minecraft-1.12.2/./crash-reports/crash-2018-07-15_21.20.59-server.txt`

```
---- Minecraft Crash Report ----
// You're mean.

Time: 7/15/18 9:20 PM
Description: Exception in server tick loop

java.lang.NoSuchMethodError: net.minecraftforge.fml.common.registry.GameRegistry.registerTileEntity(Ljava/lang/Class;Lnet/minecraft/util/ResourceLocation;)V
	at net.shadowmage.ancientwarfare.structure.block.AWStructuresBlockLoader.registerTile(AWStructuresBlockLoader.java:72)
	at net.shadowmage.ancientwarfare.structure.block.AWStructuresBlockLoader.register(AWStructuresBlockLoader.java:50)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_20_AWStructuresBlockLoader_register_Register.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus$1.invoke(EventBus.java:143)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:179)
	at net.minecraftforge.registries.GameData.fireRegistryEvents(GameData.java:734)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:604)
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:98)
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:332)
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:125)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:484)
	at java.lang.Thread.run(Thread.java:748)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Linux (amd64) version 4.13.0-32-generic
	Java Version: 1.8.0_171, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 756519792 bytes (721 MB) / 1037959168 bytes (989 MB) up to 1037959168 bytes (989 MB)
	JVM Flags: 2 total; -Xmx1024M -Xms1024M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP 9.42 Powered by Forge 14.23.1.2604 13 mods loaded, 13 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State | ID                       | Version          | Source                                        | Signature                                |
	|:----- |:------------------------ |:---------------- |:--------------------------------------------- |:---------------------------------------- |
	| UCH   | minecraft                | 1.12.2           | minecraft.jar                                 | None                                     |
	| UCH   | mcp                      | 9.42             | minecraft.jar                                 | None                                     |
	| UCH   | FML                      | 8.0.99.99        | forge-1.12.2-14.23.1.2604-universal.jar       | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCH   | forge                    | 14.23.1.2604     | forge-1.12.2-14.23.1.2604-universal.jar       | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCH   | codechickenlib           | 3.2.0.345        | CodeChickenLib-1.12.2-3.2.0.345-universal.jar | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCH   | ancientwarfare           | 1.12.2-2.6.1.285 | ancientwarfare-1.12.2-2.6.1.285.jar           | None                                     |
	| UCH   | ancientwarfareautomation | 1.12.2-2.6.1.285 | ancientwarfare-1.12.2-2.6.1.285.jar           | None                                     |
	| UCH   | ancientwarfarenpc        | 1.12.2-2.6.1.285 | ancientwarfare-1.12.2-2.6.1.285.jar           | None                                     |
	| UCH   | ancientwarfarestructure  | 1.12.2-2.6.1.285 | ancientwarfare-1.12.2-2.6.1.285.jar           | None                                     |
	| UCH   | ancientwarfarevehicle    | 1.12.2-2.6.1.285 | ancientwarfare-1.12.2-2.6.1.285.jar           | None                                     |
	| UCH   | jei                      | 4.8.0.114        | jei_1.12.2-4.8.0.114.jar                      | None                                     |
	| UCH   | mercurius_updater        | 1.0              | MercuriusUpdater-1.12.2.jar                   | None                                     |
	| UCH   | harvestcraft             | 1.12.2a          | Pam's HarvestCraft 1.12.2a.jar                | None                                     |

	Loaded coremods (and transformers):
	Profiler Position: N/A (disabled)
	Is Modded: Definitely; Server brand changed to 'fml,forge'
	Type: Dedicated Server (map_server.txt)
```

Forge: 
- forge-1.12.2-14.23.4.2738-universal.jar

Mods:
- CodeChickenLib-1.12.2-3.2.0.345-universal.jar
- Pam's HarvestCraft 1.12.2a.jar
- ancientwarfare-1.12.2-2.6.1.285.jar
- jei_1.12.2-4.8.0.114.jar

Any ideas?

**Comment by P3pp3rF1y** [comment]
Just saw this one from Mohawky who's helping with new structure pack. You just need to update the forge and that should fix it. Let me know if that helped.

**Comment by wadewegner** [comment]
I did update Forge. 1.12.2-14.23.4.2738 is the latest, yes?

**Comment by P3pp3rF1y** [comment]
However that log shows 14.23.1.2604 , or are you getting a different exception now?

**Comment by wadewegner** [comment]
Oh, you might be right. My bad, let me try again.

**Comment by wadewegner** [comment]
Yeah, apologies. I had gotten the latest but ran the previous. Sorry. Thank you!

**Comment by P3pp3rF1y** [comment]
No problem at all.
